Daemon/Daemon/PluginBase.m

```diff
@@ -76,27 +76,25 @@ -(BOOL)isMatch:(File*)file
     path = file.destinationPath;
     
     //match?
-    [self.regexes enumerateObjectsUsingBlock:^(NSRegularExpression* regex, NSUInteger index, BOOL * _Nonnull stop) {
-        
+    [self.regexes enumerateObjectsWithOptions:NSEnumerationConcurrent usingBlock:^(NSRegularExpression* _Nonnull regex, NSUInteger idx, BOOL * _Nonnull stop) {
```

Comment:
`enumerateObjectsUsingBlock:` was blocking the current thread while evaluating regular expressions against the given file path. `indexOfObjectWithOptions:NSEnumerationConcurrent` will block the current thread, but also would fire multiple threads for evaluating regular expressions on different CPU cores at the same time.

Might work, might not work. For your consideration.